instrumentation/vertx/vertx-redis-client-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/redis/VertxRedisClientRequest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.vertx.v4_0.redis;
+
+import io.vertx.core.net.NetSocket;
+import io.vertx.redis.client.impl.RedisURI;
+import java.util.List;
+import java.util.Locale;
+
+public final class VertxRedisClientRequest {
+  private final String command;
+  private final List<byte[]> args;
+  private final RedisURI redisUri;
+  private final NetSocket netSocket;
+
+  public VertxRedisClientRequest(
+      String command, List<byte[]> args, RedisURI redisUri, NetSocket netSocket) {
+    this.command = command.toUpperCase(Locale.ROOT);
+    this.args = args;
+    this.redisUri = redisUri;
+    this.netSocket = netSocket;
+  }
+
+  public String getCommand() {
+    return command;
+  }
+
+  public List<byte[]> getArgs() {
+    return args;
+  }
+
+  public String getUser() {
+    return redisUri.user();
+  }
+
+  public String getDatabase() {
+    Integer select = redisUri.select();
+    return select != null ? select.toString() : null;
```

Comment:
it looks like we could/should be using `db.redis.database_index` (https://github.com/open-telemetry/semantic-conventions/blob/main/docs/database/redis.md#semantic-conventions-for-redis)